modular_skyrat/modules/modular_implants/code/nifsofts/soulcatcher.dm

```diff
@@ -0,0 +1,86 @@
+/obj/item/disk/nifsoft_uploader/soulcatcher
+	name = "Soulcatcher"
+	loaded_nifsoft = /datum/nifsoft/soulcatcher
+
+/datum/nifsoft/soulcatcher
+	name = "Soulcatcher"
+	program_desc = "Holds souls" // PLACEHOLDER
```

Comment:
The 'Soulcatcher' coreware is a near-complete upgrade of the nanomachine systems in a NIF, meant for one purpose; supposedly, channeling the dead. This upgrade, in truth, functions as a Resonance Simulation Device; an RSD for short, an instrument capable of hosting someone's consciousness, context or otherwise.

'Resonance,' a term for consciousness independent of any one specific body, was discovered in the early 2500s by researchers Yun-Seo Jin and Kamakshi Padmanabhan, coining what is now called 'Jin-Padmanabhan Resonance,' or 'JP/Soul Resonance.' This 'resonance' is a byproduct of neural activity that gives a sophont its consciousness, its sense of continuation, and their 'I am me.' It is, most notably, replicable.

The earliest RSDs were massive machines, drawing incredible power and utilizing bleeding-edge, clunky software to 'play' someone's Resonance at 1:1 accuracy with their original brain. However, two problems arose. One, Resonance is something that requires 'context;' the memories and associations and identity essentially etched into the nerves or hardware of an organic or synthetic being respectively, the consciousness otherwise having no sense of memory or self. At first, complete deepscans of the brain were needed. Two, outside of strange cases; primarily those concerning synthetics, slimes, and other gestalt consciousnesses, it is extremely difficult to have a person's Resonance 'playing' on multiple 'instruments' at once.

The first portable RSD, or Soulcatcher, was developed by the Spider Clan. These were initially designed for the captive interrogation of a person's consciousness without having to worry about the struggling of their body, and for dead or aging members of the mysterious group of orbital shinobi to be able to guide field operatives. These Soulcatchers are the main instrument to play Resonance, but recent advances in medical science have been leading to more. 

Occasionally, it is known for unusual sources of 'wild' Resonance, called Phantoms or Engrams, to end up inside of the nearest Soulcatcher, a key finding its own lock; thought to be the dead's last hurrah, though known to often be the result of well-tangled, anomalous software quirks still being investigated. Much as how some people intentionally become stable Engrams to achieve digital immortality, such as the witches of the Altspace Coven, it is possible for others to become wild Phantoms by hacking their way into a person's Soulcatcher remotely.